src/fs/filesystem.ts

```diff
@@ -273,17 +281,34 @@ export class FileSystem implements UnixTree {
   }
 
 
+  // PUBLICIZE
+  // ---------
+
   /**
-   * Ensures the latest version of the file system is added to IPFS and returns the root CID.
+   * Ensures the latest version of the file system is added to IPFS,
+   * updates your data root, and returns the root CID.
    */
-  async publicize(): Promise<CID> {
+  async publicise(): Promise<CID> {
     const cid = await this.root.put()
+    const proofs = Array.from(Object.entries(this.proofs))
+
+    this.proofs = {}
 
-    this.syncHooks.forEach(hook => hook(cid))
+    proofs.forEach(([_, proof]) => {
```

Comment:
Totally good with putting off the UCAN 0.4 changes until next PR :ok_hand:  but I think we should adjust this logic slightly in the meantime. With UCAN 0.4, we're going to collate all the required proofs into 1 UCAN & then send one "Update Data Root" request to the server (not one per proof). Right now we're sending 1 request _per proof_. They'll get approved (because the server doesn't have the correct parsing code yet), but there's no real sense in bombarding it with multiple requests to update the same CID. Let me know if I'm misunderstanding the purpose of calling the synchooks once per proof!

tl;dr all good, but let's just call the sync hooks once per `publicise`

Reply:
> there's no real sense in bombarding it with multiple requests to update the same CID.
Totally, that's why adjusted the auth lobby to only return one UCAN.
So it'll only use that one UCAN here, and thus only make one request.

Reply:
ah got it. Auth lobby only sends 1 UCAN & we're deduping this so only sending one request. makes sense!